### PR TITLE
[Maintenance] Remove obsolete docs, restructure for better UX - GT-314

### PR DESCRIPTION
**Pull request by nikita-vanyasin**
- move docs/design/features to docs/features for better structure
- add generator for features index (docs/features/README.md)
- add generator for API reference index (docs/api/README.md)
- remove obsolete docs
- update old docs with new information (e.g. there is ArangoDeployment.Status.State field anymore)
- move guides from `design` to `how-to` section

Check result here: https://github.com/arangodb/kube-arangodb/tree/6d452c07f33bbc61d3fe753602b8d7db43d1ac80/docs

In next PRs we are going to move docs from main documentation website to this repo.

### CODE REVIEW

File: docs/design/maintenance.md
Comment (jwierzbo) [inline docs/design/maintenance.md:1]: why this files has been removed?

File: docs/bare-metal.md
Comment (jwierzbo) [inline docs/bare-metal.md:1]: @ajanikow shouldn't we keep that doc?

File: docs/design/maintenance.md
Comment (nikita-vanyasin) [reply docs/design/maintenance.md:1]: @jwierzbo Incorrect git history for some reason. It was moved to How-to

File: docs/bare-metal.md
Comment (nikita-vanyasin) [reply docs/bare-metal.md:1]: @jwierzbo 90% of this doc is about setting up a kubernetes on bare-metal. Rest 10% about operator is duplication of what he have in main README. I think we should have only kube-arangodb docs in this repo, so old doc is removed.

File: docs/design/maintenance.md
Comment (jwierzbo) [reply docs/design/maintenance.md:1]: yeah I see: https://github.com/arangodb/kube-arangodb/blob/6d452c07f33bbc61d3fe753602b8d7db43d1ac80/docs/how-to/maintenance.md

File: docs/bare-metal.md
Comment (nikita-vanyasin) [reply docs/bare-metal.md:1]: I can move it to confluence just in case if we ever need that

**Review by jwierzbo — APPROVED** [review]

**Pull request merged**